src/Aspire.Hosting.MongoDB/MongoDBServerResource.cs

```diff
@@ -10,20 +10,95 @@ namespace Aspire.Hosting.ApplicationModel;
 public class MongoDBServerResource(string name) : ContainerResource(name), IResourceWithConnectionString
 {
     internal const string PrimaryEndpointName = "tcp";
-
+    private const string DefaultUserName = "admin";
+    private const string DefaultAuthenticationDatabase = "admin";
+    private const string DefaultAuthenticationMechanism = "SCRAM-SHA-256";
+    
     private EndpointReference? _primaryEndpoint;
 
+    /// <summary>
+    /// Initialize a resource that represents a MongoDB container.
+    /// </summary>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="userNameParameter">A parameter that contains the MongoDb server user name, or <see langword="null"/> to use a default value.</param>
+    /// <param name="passwordParameter">A parameter that contains the MongoDb server password.</param>
+    /// <param name="authenticationMechanism">A parameter that contains the MongoDb server authentication mechanism, or <see langword="null"/> to use a default value.</param>
+    public MongoDBServerResource(string name, ParameterResource? userNameParameter, ParameterResource? passwordParameter, ParameterResource? authenticationMechanism) : this(name)
```

Comment:
```suggestion
    public MongoDBServerResource(string name, ParameterResource? userNameParameter, ParameterResource passwordParameter, ParameterResource? authenticationMechanism) : this(name)
```

The rest of the resources that take a password parameter aren't nullable.